File: Warm-Up Challenges/SockMerchant.c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

int main() {
int n;
    scanf("%d",&n);
    int a[n],i,j,s=0;
    for(i=0;i<n;i++)
        {
        scanf("%d",&a[i]);
    }
    
    for(i=0;i<n-1;i++)
        {
        for(j=i+1;j<n;j++)
            {
            if(a[i]==a[j]&&a[i]!=0&&a[j]!=0)
               { s++;
                a[i]=0;
                a[j]=0;}
        }
    }printf("%d",s);
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */    
    return 0;
}

File: Warm-Up Challenges/SockMerchant.js
function processData(input) {
    //Enter your code here
  var pInput = input.split('\n');
  var n = pInput[0];
  var pInput = pInput[1].split(' ');
  var oddItems = {};
  pInput.map(function (i) {
    if(oddItems[i]) {
      delete oddItems[i];
    } else {
      oddItems[i] = true;
    }
  })
  console.log((n - Object.keys(oddItems).length)/2);
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});